packages/sdk/src/evm/common/any-evm-utils/getInitBytecodeWithSalt.ts

```diff
@@ -13,8 +13,9 @@ import { getSaltHash } from "./getSaltHash";
 export function getInitBytecodeWithSalt(
   bytecode: string,
   encodedArgs: BytesLike,
+  salt?: string,
 ): string {
-  const saltHash = getSaltHash(bytecode);
+  const saltHash = salt ? getSaltHash(salt) : getSaltHash(bytecode);
```

Comment:
don't process salt if its being passed in, just use it straight up so there's no surprises for the caller